docs(release.md): update steps (#1684)
falcosecurity/falco: RELEASE.md

    | `docker pull docker.io/falcosecurity/falco-driver-loader:x.y.z`             |
    | `docker pull docker.io/falcosecurity/falco-no-driver:x.y.z`                 |

    <changelog>

>     <!-- Substitute changelog with the one generated by [rn2md](https://github.com/leodido/rn2md) -->

Reviewer: ```suggestion
    <!-- Substitute <changelog> with the one generated by [rn2md](https://github.com/leodido/rn2md) -->
```

- Publish a blog on [Falco website](https://github.com/falcosecurity/falco-website) ([example](https://github.com/falcosecurity/falco-website/blob/master/content/en/blog/falco-0-28-1.md))
- Send an announcement to cncf-falco-dev@lists.cncf.io (plain text, please)
- Let folks in the slack #falco channel know about a new release came out

> - If and only if, this release introduces a **new minor version**, [archive a snapshot of the Falco website](https://github.com/falcosecurity/falco-website/blob/master/release.md#documentation-versioning)

Reviewer: ```suggestion
- IFF the on going release introduces a **new minor version**, [archive a snapshot of the Falco website](https://github.com/falcosecurity/falco-website/blob/master/release.md#documentation-versioning)
```
